### PR TITLE
[Resolves 21] add list-user function

### PR DESCRIPTION
**Pull request by cascadianblue**
Add a search function (user-list).

This PR also contains a bugfix for create-user, where first name and last name were not getting added to users.

### CODE REVIEW

File: unit_tests/test_jc_api_v1.py [hunk 1]
```diff
@@ -138,6 +138,28 @@ def test_get_user_no_id(self, mock_systemusers_list):
...
+                firstname='David',
+                email='david@david.net'
+            ),
+        ]
+        mock_search_systemusers_post.return_value = Systemuserslist(results=users, total_count=len(users))
+
+        api1 = JumpcloudApiV1("fake_key_123")
+        user_search = api1.search_users({'firstname': 'David'})
+        assert (
+            user_search == [user.to_dict() for user in users]
```
Comment (zaro0508) [inline unit_tests/test_jc_api_v1.py:155]: could we create a test for no users found?

File: unit_tests/test_jc_api_v1.py [hunk 2]
```diff
@@ -138,6 +138,28 @@ def test_get_user_no_id(self, mock_systemusers_list):
         with pytest.raises(SystemUserNotFoundError):
             api1.get_user('angela')
 
+    @patch.object(jcapiv1.SearchApi, 'search_systemusers_post')
+    def test_search_users(self, mock_search_systemusers_post):
```
Comment (zaro0508) [inline unit_tests/test_jc_api_v1.py:142]: could we create a unit test for the AND condition (first & last name)?

Comment (cascadianblue) [reply unit_tests/test_jc_api_v1.py:142]: Good idea. I think I'll add a step to the integration tests that does the same as well

Comment (cascadianblue) [reply unit_tests/test_jc_api_v1.py:142]: Sorry, I'm recovering from a severe case of vacation brain... we already have one such integration test. I'll just make the unit test.

Comment (zaro0508) [reply unit_tests/test_jc_api_v1.py:142]: Generally, an effective unit test is preferable over an integration test.  In this case i think a unit test is enough to test this functionality.

**Pull request merged**